diff --git a/core/image.cpp b/core/image.cpp
index ff4937527f..7134051802 100644
--- a/core/image.cpp
+++ b/core/image.cpp
@@ -2040,11 +2040,6 @@ Image::AlphaMode Image::detect_alpha() const {
 }
 
 Error Image::load(const String &p_path) {
-#ifdef DEBUG_ENABLED
-	if (p_path.begins_with("res://") && ResourceLoader::exists(p_path)) {
-		WARN_PRINT("Loaded resource as image file, this will not work on export: '" + p_path + "'. Instead, import the image file as an Image resource and load it normally as a resource.");
-	}
-#endif
 	return ImageLoader::load_image(p_path, this);
 }
 
